build.cake

```diff
@@ -223,59 +223,89 @@ Task("Externals-Uwp")
     CleanDirectory("externals/uwp");
     EnsureDirectoryExists("Assets/AppCenter/Plugins/WSA/");
     // Download the nugets. We will use these to extract the dlls
-    foreach (var module in AppCenterModules)
-    {
-        if (module.Moniker == "Distribute")
-        {
-            Warning("Skipping 'Distribute' for UWP.");
-            continue;
-        }
-        if (module.Moniker == "Crashes")
-        {
-            Warning("Skipping 'Crashes' for UWP.");
-            continue;
-        }
-        Information("Downloading " + module.DotNetModule + "...");
-        // Download nuget package
-        var nupkgPath = GetNuGetPackage(module.DotNetModule, UwpSdkVersion);
-
-        var tempContentPath = "externals/uwp/" + module.Moniker + "/";
-        DeleteDirectoryIfExists(tempContentPath);
-        // Unzip into externals/uwp/
-        Unzip(nupkgPath, tempContentPath);
-        // Delete the package
-        DeleteFiles(nupkgPath);
-
-        var contentPathSuffix = "lib/uap10.0/";
-
-        // Prepare destination
-        var destination = "Assets/AppCenter/Plugins/WSA/" + module.Moniker + "/";
-        EnsureDirectoryExists(destination);
-        DeleteFiles(destination + "*.dll");
-        DeleteFiles(destination + "*.winmd");
-
-        // Deal with any native components
-        if (module.UWPHasNativeCode)
-        {
-            foreach (var arch in module.NativeArchitectures)
-            {
-                var dest = "Assets/AppCenter/Plugins/WSA/" + module.Moniker + "/" + arch.ToString().ToUpper() + "/";
-                EnsureDirectoryExists(dest);
-                var nativeFiles = GetFiles(tempContentPath + "runtimes/" + "win10-" + arch + "/native/*");
-                DeleteFiles(dest + "*.dll");
-                MoveFiles(nativeFiles, dest);
-            }
-
-            // Use managed runtimes from one of the architecture for all architectures.
-            // Even though they are architecture dependent, Unity converts
-            // them to AnyCPU automatically
-            contentPathSuffix = "runtimes/win10-" + module.NativeArchitectures[0] + "/" + contentPathSuffix;
-        }
-
-        // Move the files to the proper location
-        var files = GetFiles(tempContentPath + contentPathSuffix + "*");
-        MoveFiles(files, destination);
-    }
+    if (EnvironmentVariable("NUGET_PRIVATE") != "Yes") {
+		var packageSource = "https://www.nuget.org/api/v2/";
```

Comment:
Have you considered about moving this two parts to separate functions to make the overall method size a smaller + the name of the func will show the intent in a more clear and concise way?